pyupgrade/_plugins/unpack_list_comprehension.py

```diff
@@ -0,0 +1,41 @@
+import ast
+import sys
+from typing import Iterable
+from typing import List
+from typing import Tuple
+
+from tokenize_rt import Offset
+from tokenize_rt import Token
+
+from pyupgrade._ast_helpers import ast_to_offset
+from pyupgrade._data import register
+from pyupgrade._data import State
+from pyupgrade._data import TokenFunc
+from pyupgrade._token_helpers import find_closing_bracket
+
+
+def _replace_list_comprehension(i: int, tokens: List[Token]) -> None:
+    if sys.version_info < (3, 8):  # pragma: no cover (py38+)
+        start = i - 1
+        while not (tokens[start].name == 'OP' and tokens[start].src == '['):
+            start -= 1
+    else:  # pragma: no cover (<py38)
+        start = i
```

Comment:
I think there might be a helper for this somewhere in pyupgrade already? -- maybe try the set / dict comp rewrites?

Reply:
I've had a look at `_arg_token_index` and `victims` from `pyupgrade._token_helpers`, but it doesn't seem so easy to adapt them here.

They cover the case when the node yielded contains a function (such as `set`) which contains a comprehension. Here, on the other hand, the yielded node is the comprehension itself, and in <py38 I need to go _backwards_ to find the starting bracket (rather than forward to find the first element).

I'll take a closer look / have a harder think anyway

Reply:
Having looked at this further, I can't figure out how to use `victims` here...any hints? Else I'll get back to it later in the week, bit stuck on this

Reply:
it's probably fine as is